internal/resolver/config_selector.go

```diff
@@ -29,8 +29,9 @@ import (
 
 // ConfigSelector controls what configuration to use for every RPC.
 type ConfigSelector interface {
-	// Selects the configuration for the RPC.
-	SelectConfig(RPCInfo) *RPCConfig
+	// Selects the configuration for the RPC, or terminates it using the error
+	// (converted to a status error if necessary).
```

Comment:
> converted to a status error if necessary

Why does this mean?
 - The returned error should always be a status error (so it's the implementation's job to convert)
 - Or, the returned error is not a status error (so it's the caller's job to convert)?

Reply:
Clarified